<a href="https://colab.research.google.com/github/IrvanAlif/checkpoint-1_AVD/blob/main/checkpoint%202.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Business Understanding**


# **Business objective**

Tujuan utama proyek ini adalah untuk memahami perilaku pengguna mobile dalam penggunaan perangkat mobile guna mendukung peningkatan layanan digital dan meningkatkan pengalaman pengguna. Dengan menganalisis pola penggunaan aplikasi, waktu layar harian, serta kategori aplikasi yang paling sering digunakan,dengan itu perusahaan dapat:

1. meningkatkan layanan/fitur: Menambahkan fitur sesuai dengan kebutuhan pelanggan

2. Memprediksi penggunaan layanan/fitur: merekomendasika fitur sesui dengan kebiasaan pengguna

3. Meningkatkan personalisasi konten: Memberikan rekommendasi yang serupa dengan yang biasa di nikmati pelanggan

# **Assess Situation**

Situasi bisnis yang mendasari analisis ini adalah semakin meningkatnya ketergantungan masyarakat terhadap perangkat mobile dalam aktivitas sehari-hari. Pengguna menggunakan berbagai macam aplikasi untuk melakukan komunikasi, mendapatkan hiburan, meningkatkan produktivitas, dan kebutuhan lainnya. Dalam persaingan yang semakin kompetitif, perusahaan harus memahami pola penggunaan mobile untuk menyesuaikan strategi bisnis dengan tren penggunaan aplikasi mobile, memahami pola perilaku pengguna untuk menciptakan fitur yang lebih baik, serta meningkatkan pengalaman pengguna untuk meningkatkan fitur-fitur yang sudah ada









# **Data Mining Goals**

Mengidentifikasi segmentasi pengguna berdasarkan pola penggunaan aplikasi, durasi screen time, serta karakteristik pengguna. Selain itu, analisis ini bertujuan untuk menemukan hubungan antara berbagai faktor, seperti usia dengan jenis aplikasi yang paling sering digunakan, serta memprediksi pengunaan aplikasi tertentu dalam jangka panjang. Proyek ini juga bertujuan untuk mengenali kelompok yang berpotensi mendapatkan manfaat dari layanan ini, serta menemukan pola anomali atau perilaku tidak biasa, misalnya lonjakan penggunaan aplikasi dalam periode tertentu karena tren yang sedang berlangsung. Dengan mencapai tujuan ini, perusahaan dapat meningkatkan efektivitas layanan digital

# **project plan**

Proyek ini akan dilakukan dalam beberapa tahap yang dimulai dengan pengumpulan data untuk memahami pola penggunaan aplikasi oleh pengguna. Setelah itu, data akan diproses dan dibersihkan untuk memastikan kualitasnya sebelum dianalisis lebih lanjut. Analisis data akan dilakukan dengan metode data mining untuk mengidentifikasi pola penggunaan, segmentasi pengguna, serta faktor-faktor yang memengaruhi pemilihan aplikasi. Hasil dari analisis ini akan digunakan untuk menyusun strategi yang dapat meningkatkan pengalaman pengguna. Tahap akhir proyek meliputi evaluasi hasil, penyusunan laporan, serta rekomendasi dalam betuk data yang dapat diterapkan oleh perusahaan dalam pengembangan layanan mereka.

# **Data Understanding**


link kaggle
https://www.kaggle.com/datasets/bhadramohit/smartphone-usage-and-behavioral-dataset

## **Data**

In [48]:
import pandas as pd

In [56]:
df = pd.read_csv("/content/mobile_usage_behavioral_analysis.csv")

df

User_ID  Age  Gender  Total_App_Usage_Hours  Daily_Screen_Time_Hours  \
0          1   56    Male                   2.61                     7.15   
1          2   46    Male                   2.13                    13.79   
2          3   32  Female                   7.28                     4.50   
3          4   25  Female                   1.20                     6.29   
4          5   38    Male                   6.31                    12.59   
..       ...  ...     ...                    ...                      ...   
995      996   41  Female                   3.58                    12.46   
996      997   32    Male                   9.65                    10.51   
997      998   46  Female                  11.68                    10.06   
998      999   25    Male                   3.89                    10.78   
999     1000   22  Female                   3.99                     7.05   

     Number_of_Apps_Used  Social_Media_Usage_Hours  \
0                     24                      4.43   
1                     18                      4.67   
2                     11                      4.58   
3                     21                      3.18   
4                     14                      3.15   
..                   ...                       ...   
995                   13                      4.12   
996                   11                      1.02   
997                    9                      2.35   
998                   15                      4.00   
999                    4                      0.90   

     Productivity_App_Usage_Hours  Gaming_App_Usage_Hours     Location  
0                            0.55                    2.40  Los Angeles  
1                            4.42                    2.43      Chicago  
2                            1.71                    2.83      Houston  
3                            3.42                    4.58      Phoenix  
4                            0.13                    4.00     New York  
..                            ...                     ...          ...  
995                          2.44                    1.33  Los Angeles  
996                          1.02                    3.84  Los Angeles  
997                          3.13                    1.03      Phoenix  
998                          1.79                    1.61      Phoenix  
999                          4.98                    2.33      Phoenix  

[1000 rows x 10 columns]

## **ringkasan informasi**

In [ ]:
df.info(), df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 10 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   User_ID                       1000 non-null   int64  
 1   Age                           1000 non-null   int64  
 2   Gender                        1000 non-null   object 
 3   Total_App_Usage_Hours         1000 non-null   float64
 4   Daily_Screen_Time_Hours       1000 non-null   float64
 5   Number_of_Apps_Used           1000 non-null   int64  
 6   Social_Media_Usage_Hours      1000 non-null   float64
 7   Productivity_App_Usage_Hours  1000 non-null   float64
 8   Gaming_App_Usage_Hours        1000 non-null   float64
 9   Location                      1000 non-null   object 
dtypes: float64(5), int64(3), object(2)
memory usage: 78.3+ KB


(None,
    User_ID  Age  Gender  Total_App_Usage_Hours  Daily_Screen_Time_Hours  \
 0        1   56    Male                   2.61                     7.15   
 1        2   46    Male                   2.13                    13.79   
 2        3   32  Female                   7.28                     4.50   
 3        4   25  Female                   1.20                     6.29   
 4        5   38    Male                   6.31                    12.59   
 
    Number_of_Apps_Used  Social_Media_Usage_Hours  \
 0                   24                      4.43   
 1                   18                      4.67   
 2                   11                      4.58   
 3                   21                      3.18   
 4                   14                      3.15   
 
    Productivity_App_Usage_Hours  Gaming_App_Usage_Hours     Location  
 0                          0.55                    2.40  Los Angeles  
 1                          4.42                    2.43      Chicago  

1. Dataset memiliki 1000 baris dan 10 kolom.
2. Tipe data pada setiap kolom:

    a. Integer (int64): User_ID, Age, Number_of_Apps_Used

    b.  Float (float64): Total_App_Usage_Hours, Daily_Screen_Time_Hours, Social_Media_Usage_Hours, Productivity_App_Usage_Hours, Gaming_App_Usage_Hours

    c. String (object): Gender, Location
    
3. Tidak ada nilai yang hilang (Non-Null Count sama dengan jumlah total baris).

## **menampilkan statistik**

In [ ]:
df.describe()

User_ID          Age  Total_App_Usage_Hours  \
count  1000.000000  1000.000000            1000.000000   
mean    500.500000    38.745000               6.405670   
std     288.819436    12.186734               3.134855   
min       1.000000    18.000000               1.000000   
25%     250.750000    28.000000               3.590000   
50%     500.500000    40.000000               6.455000   
75%     750.250000    50.000000               9.122500   
max    1000.000000    59.000000              11.970000   

       Daily_Screen_Time_Hours  Number_of_Apps_Used  Social_Media_Usage_Hours  \
count              1000.000000           1000.00000               1000.000000   
mean                  7.696310             16.64700                  2.456330   
std                   3.714187              7.61961                  1.439525   
min                   1.010000              3.00000                  0.000000   
25%                   4.530000             10.00000                  1.200000   
50%                   7.880000             17.00000                  2.445000   
75%                  10.910000             23.00000                  3.672500   
max                  14.000000             29.00000                  4.990000   

       Productivity_App_Usage_Hours  Gaming_App_Usage_Hours  
count                   1000.000000             1000.000000  
mean                       2.495270                2.475410  
std                        1.443392                1.450362  
min                        0.000000                0.010000  
25%                        1.282500                1.220000  
50%                        2.435000                2.455000  
75%                        3.710000                3.782500  
max                        5.000000                5.000000


* Rata-rata usia pengguna sekitar 38,75 tahun, dengan rentang 18 hingga 59 tahun.

* Rata-rata waktu penggunaan aplikasi harian sekitar 6,4 jam, dengan beberapa pengguna yang hanya 1 jam, sementara yang paling tinggi hampir 12 jam per hari.

* Jumlah aplikasi yang digunakan bervariasi, dari 3 hingga 29 aplikasi, dengan median sekitar 17 aplikasi per hari.

* Rata-rata waktu untuk media sosial sekitar 2,45 jam, produktivitas 2,49 jam, dan gaming 2,47 jam.


## **menghitung korelasi**

In [ ]:
df_numeric = df.select_dtypes(include=['number'])

correlation_matrix = df_numeric.corr()

print(correlation_matrix)

                               User_ID       Age  Total_App_Usage_Hours  \
User_ID                       1.000000 -0.028576              -0.010570   
Age                          -0.028576  1.000000              -0.003319   
Total_App_Usage_Hours        -0.010570 -0.003319               1.000000   
Daily_Screen_Time_Hours       0.020941 -0.017639               0.002948   
Number_of_Apps_Used          -0.001214 -0.002081               0.041493   
Social_Media_Usage_Hours     -0.002608 -0.008324               0.019330   
Productivity_App_Usage_Hours  0.002207  0.005848              -0.009983   
Gaming_App_Usage_Hours       -0.003514 -0.012969              -0.067770   

                              Daily_Screen_Time_Hours  Number_of_Apps_Used  \
User_ID                                      0.020941            -0.001214   
Age                                         -0.017639            -0.002081   
Total_App_Usage_Hours                        0.002948             0.041493   
Daily_Screen

**Korelasi Terkuat:**

*   "Social_Media_Usage_Hours" dan "Productivity_App_Usage_Hours" → -0.0765
(negatif paling kuat)

* "Total_App_Usage_Hours" dan "Gaming_App_Usage_Hours" → -0.0677
Meski kecil, ini menunjukkan bahwa semakin lama seseorang menggunakan media sosial, semakin sedikit waktu untuk aplikasi produktivitas.

**Korelasi Terlemah:**

* "Age" dan "Total_App_Usage_Hours" → -0.0033

* "Gaming_App_Usage_Hours" dan "Daily_Screen_Time_Hours" → -0.0129
Hampir tidak ada hubungan antara umur dan total waktu penggunaan aplikasi, serta screen time dengan gaming.

# **Data Cleaning**

## **Missing Values**

In [ ]:
print((df.isna().sum() / len(df)) * 100)

User_ID                         0.0
Age                             0.0
Gender                          0.0
Total_App_Usage_Hours           0.0
Daily_Screen_Time_Hours         0.0
Number_of_Apps_Used             0.0
Social_Media_Usage_Hours        0.0
Productivity_App_Usage_Hours    0.0
Gaming_App_Usage_Hours          0.0
Location                        0.0
dtype: float64


* Karena semua hasilnya 0.0, berarti tidak ada data yang kosong di dalam DataFrame.

## Duplicated Values

In [ ]:
df[df.duplicated()]

Empty DataFrame
Columns: [User_ID, Age, Gender, Total_App_Usage_Hours, Daily_Screen_Time_Hours, Number_of_Apps_Used, Social_Media_Usage_Hours, Productivity_App_Usage_Hours, Gaming_App_Usage_Hours, Location]
Index: []

Tidak ada data duplicate

In [ ]:
df

User_ID  Age  Gender  Total_App_Usage_Hours  Daily_Screen_Time_Hours  \
0          1   56    Male                   2.61                     7.15   
1          2   46    Male                   2.13                    13.79   
2          3   32  Female                   7.28                     4.50   
3          4   25  Female                   1.20                     6.29   
4          5   38    Male                   6.31                    12.59   
..       ...  ...     ...                    ...                      ...   
995      996   41  Female                   3.58                    12.46   
996      997   32    Male                   9.65                    10.51   
997      998   46  Female                  11.68                    10.06   
998      999   25    Male                   3.89                    10.78   
999     1000   22  Female                   3.99                     7.05   

     Number_of_Apps_Used  Social_Media_Usage_Hours  \
0                     24                      4.43   
1                     18                      4.67   
2                     11                      4.58   
3                     21                      3.18   
4                     14                      3.15   
..                   ...                       ...   
995                   13                      4.12   
996                   11                      1.02   
997                    9                      2.35   
998                   15                      4.00   
999                    4                      0.90   

     Productivity_App_Usage_Hours  Gaming_App_Usage_Hours     Location  
0                            0.55                    2.40  Los Angeles  
1                            4.42                    2.43      Chicago  
2                            1.71                    2.83      Houston  
3                            3.42                    4.58      Phoenix  
4                            0.13                    4.00     New York  
..                            ...                     ...          ...  
995                          2.44                    1.33  Los Angeles  
996                          1.02                    3.84  Los Angeles  
997                          3.13                    1.03      Phoenix  
998                          1.79                    1.61      Phoenix  
999                          4.98                    2.33      Phoenix  

[1000 rows x 10 columns]

## **outliers**

In [10]:
results = []

cols = df.select_dtypes(include=['float64', 'int64'])

for col in cols:
  q1 = df[col].quantile(0.25)
  q3 = df[col].quantile(0.75)
  iqr = q3 - q1
  lower_bound = q1 - 1.5*iqr
  upper_bound = q3 + 1.5*iqr
  outliers = df[(df[col] < lower_bound) | (df[col] > upper_bound)]
  percent_outliers = (len(outliers)/len(df))*100
  results.append({'Kolom': col, 'Persentase Outliers': percent_outliers})

# Dataframe dari list hasil
results_df = pd.DataFrame(results)
results_df.set_index('Kolom', inplace=True)
results_df = results_df.rename_axis(None, axis=0).rename_axis('Kolom', axis=1)

# Tampilkan dataframe
display(results_df)

Kolom                         Persentase Outliers
User_ID                                       0.0
Age                                           0.0
Total_App_Usage_Hours                         0.0
Daily_Screen_Time_Hours                       0.0
Number_of_Apps_Used                           0.0
Social_Media_Usage_Hours                      0.0
Productivity_App_Usage_Hours                  0.0
Gaming_App_Usage_Hours                        0.0

Tidak ada nilai yang berada di luar batas IQR (Interquartile Range). Dengan kata lain, semua data dalam kolom numerik berada dalam rentang yang dianggap normal menurut metode IQR.

## **Inconsistent Values**

Dataset yang saya pilih tidak ada nilai yang tidak konsisten

In [50]:
df

User_ID  Age  Gender  Total_App_Usage_Hours  Daily_Screen_Time_Hours  \
0          1   56    Male                   2.61                     7.15   
1          2   46    Male                   2.13                    13.79   
2          3   32  Female                   7.28                     4.50   
3          4   25  Female                   1.20                     6.29   
4          5   38    Male                   6.31                    12.59   
..       ...  ...     ...                    ...                      ...   
995      996   41  Female                   3.58                    12.46   
996      997   32    Male                   9.65                    10.51   
997      998   46  Female                  11.68                    10.06   
998      999   25    Male                   3.89                    10.78   
999     1000   22  Female                   3.99                     7.05   

     Number_of_Apps_Used  Social_Media_Usage_Hours  \
0                     24                      4.43   
1                     18                      4.67   
2                     11                      4.58   
3                     21                      3.18   
4                     14                      3.15   
..                   ...                       ...   
995                   13                      4.12   
996                   11                      1.02   
997                    9                      2.35   
998                   15                      4.00   
999                    4                      0.90   

     Productivity_App_Usage_Hours  Gaming_App_Usage_Hours     Location  
0                            0.55                    2.40  Los Angeles  
1                            4.42                    2.43      Chicago  
2                            1.71                    2.83      Houston  
3                            3.42                    4.58      Phoenix  
4                            0.13                    4.00     New York  
..                            ...                     ...          ...  
995                          2.44                    1.33  Los Angeles  
996                          1.02                    3.84  Los Angeles  
997                          3.13                    1.03      Phoenix  
998                          1.79                    1.61      Phoenix  
999                          4.98                    2.33      Phoenix  

[1000 rows x 10 columns]

  User_ID  Daily_Usage_Minutes  Apps_Used  Total_Data_Consumed_MB  \
0    U001                  454         32                  963.92   
1    U002                  246          5                 4216.51   
2    U003                   71          7                 3462.58   
3    U004                  231         30                 4445.26   
4    U005                  382         16                 2384.04   

   Battery_Percentage_End  User_Category  
0                      53  Moderate User  
1                      65  Moderate User  
2                      53     Light User  
3                      97     Light User  
4                      49  Moderate User  


Hasil kode di atas menampilkan lima baris pertama dari DataFrame `df`, yang berisi data tentang penggunaan harian perangkat oleh beberapa pengguna. Berikut adalah penjelasan setiap kolomnya:

1. **`User_ID`** → ID unik untuk setiap pengguna (misalnya, `U001`, `U002`, dll.).
2. **`Daily_Usage_Minutes`** → Jumlah menit penggunaan perangkat dalam sehari.
3. **`Apps_Used`** → Jumlah aplikasi yang digunakan oleh pengguna dalam sehari.
4. **`Total_Data_Consumed_MB`** → Total data internet yang digunakan dalam satu hari (dalam MB).
5. **`Battery_Percentage_End`** → Persentase baterai yang tersisa di akhir hari.

Misalnya, baris pertama (`U001`) menunjukkan bahwa pengguna tersebut:
- Menggunakan perangkat selama **454 menit** dalam sehari.
- Menggunakan **32 aplikasi**.
- Menghabiskan **963.92 MB** data.
- Di akhir hari, sisa baterainya **53%**.


**Kategori pengguna berdasarkan Daily_Usage_Minutes:**

Light User → < 4 jam (240 menit).

Moderate User → 4 - 8 jam (240 - 480 menit).

Heavy User → > 8 jam (480 menit ke atas).

**Di temukan bahwa:**

- U001 & U005 → Menggunakan perangkat lebih dari 6 jam/hari, masuk kategori Moderate User.

- U002 → 4.10 jam, masih dalam kategori Moderate User.

- U003 & U004 → Menggunakan kurang dari 4 jam/hari, masuk kategori Light User.

- U004 memiliki 97% baterai tersisa, menunjukkan penggunaan daya yang rendah meskipun menggunakan 30 aplikasi.

## **Construct Data**

In [57]:
df

User_ID  Age  Gender  Total_App_Usage_Hours  Daily_Screen_Time_Hours  \
0          1   56    Male                   2.61                     7.15   
1          2   46    Male                   2.13                    13.79   
2          3   32  Female                   7.28                     4.50   
3          4   25  Female                   1.20                     6.29   
4          5   38    Male                   6.31                    12.59   
..       ...  ...     ...                    ...                      ...   
995      996   41  Female                   3.58                    12.46   
996      997   32    Male                   9.65                    10.51   
997      998   46  Female                  11.68                    10.06   
998      999   25    Male                   3.89                    10.78   
999     1000   22  Female                   3.99                     7.05   

     Number_of_Apps_Used  Social_Media_Usage_Hours  \
0                     24                      4.43   
1                     18                      4.67   
2                     11                      4.58   
3                     21                      3.18   
4                     14                      3.15   
..                   ...                       ...   
995                   13                      4.12   
996                   11                      1.02   
997                    9                      2.35   
998                   15                      4.00   
999                    4                      0.90   

     Productivity_App_Usage_Hours  Gaming_App_Usage_Hours     Location  
0                            0.55                    2.40  Los Angeles  
1                            4.42                    2.43      Chicago  
2                            1.71                    2.83      Houston  
3                            3.42                    4.58      Phoenix  
4                            0.13                    4.00     New York  
..                            ...                     ...          ...  
995                          2.44                    1.33  Los Angeles  
996                          1.02                    3.84  Los Angeles  
997                          3.13                    1.03      Phoenix  
998                          1.79                    1.61      Phoenix  
999                          4.98                    2.33      Phoenix  

[1000 rows x 10 columns]

In [58]:
# Menyimpan daftar kolom awal sebelum modifikasi
original_columns = df.columns.tolist()

# Fungsi untuk menentukan kategori pengguna berdasarkan Daily_Screen_Time_Hours
def categorize_user(screen_time_hours):
    if screen_time_hours < 4:
        return "Light User"
    elif 4 <= screen_time_hours <= 8:
        return "Moderate User"
    else:
        return "Heavy User"

# Menambahkan kolom 'User_Category' tanpa mengubah struktur dataset lainnya
df["User_Category"] = df["Daily_Screen_Time_Hours"].apply(categorize_user)

# Memastikan format tetap 1000 rows × 10 columns (hanya mengganti kolom terakhir dengan User_Category)
df = df[original_columns + ["User_Category"]]  # menambahn 'User_Category'

# Mengecek bentuk akhir dataset
df.shape


(1000, 11)

In [60]:
df.head(11)

User_ID  Age  Gender  Total_App_Usage_Hours  Daily_Screen_Time_Hours  \
0         1   56    Male                   2.61                     7.15   
1         2   46    Male                   2.13                    13.79   
2         3   32  Female                   7.28                     4.50   
3         4   25  Female                   1.20                     6.29   
4         5   38    Male                   6.31                    12.59   
5         6   56  Female                   3.31                     1.04   
6         7   36    Male                   4.81                     1.52   
7         8   40    Male                   9.53                     8.85   
8         9   28    Male                  11.35                    12.80   
9        10   28    Male                   1.80                     4.11   
10       11   41  Female                   6.88                     7.11   

    Number_of_Apps_Used  Social_Media_Usage_Hours  \
0                    24                      4.43   
1                    18                      4.67   
2                    11                      4.58   
3                    21                      3.18   
4                    14                      3.15   
5                    21                      2.02   
6                    24                      3.76   
7                    11                      2.66   
8                     9                      3.39   
9                    16                      2.14   
10                   21                      3.66   

    Productivity_App_Usage_Hours  Gaming_App_Usage_Hours     Location  \
0                           0.55                    2.40  Los Angeles   
1                           4.42                    2.43      Chicago   
2                           1.71                    2.83      Houston   
3                           3.42                    4.58      Phoenix   
4                           0.13                    4.00     New York   
5                           3.45                    2.26      Chicago   
6                           1.58                    3.59  Los Angeles   
7                           0.28                    2.93      Chicago   
8                           3.05                    2.46      Phoenix   
9                           1.20                    0.96     New York   
10                          1.93                    2.32      Chicago   

    User_Category  
0   Moderate User  
1      Heavy User  
2   Moderate User  
3   Moderate User  
4      Heavy User  
5      Light User  
6      Light User  
7      Heavy User  
8      Heavy User  
9   Moderate User  
10  Moderate User

User_Category berhasil di tambahkan berdasarkan Daily_Screen_Time_Hours

## **Data Reduction**

In [59]:
df.head(11)  # Menampilkan 10 baris pertama

User_ID  Age  Gender  Total_App_Usage_Hours  Daily_Screen_Time_Hours  \
0         1   56    Male                   2.61                     7.15   
1         2   46    Male                   2.13                    13.79   
2         3   32  Female                   7.28                     4.50   
3         4   25  Female                   1.20                     6.29   
4         5   38    Male                   6.31                    12.59   
5         6   56  Female                   3.31                     1.04   
6         7   36    Male                   4.81                     1.52   
7         8   40    Male                   9.53                     8.85   
8         9   28    Male                  11.35                    12.80   
9        10   28    Male                   1.80                     4.11   
10       11   41  Female                   6.88                     7.11   

    Number_of_Apps_Used  Social_Media_Usage_Hours  \
0                    24                      4.43   
1                    18                      4.67   
2                    11                      4.58   
3                    21                      3.18   
4                    14                      3.15   
5                    21                      2.02   
6                    24                      3.76   
7                    11                      2.66   
8                     9                      3.39   
9                    16                      2.14   
10                   21                      3.66   

    Productivity_App_Usage_Hours  Gaming_App_Usage_Hours     Location  \
0                           0.55                    2.40  Los Angeles   
1                           4.42                    2.43      Chicago   
2                           1.71                    2.83      Houston   
3                           3.42                    4.58      Phoenix   
4                           0.13                    4.00     New York   
5                           3.45                    2.26      Chicago   
6                           1.58                    3.59  Los Angeles   
7                           0.28                    2.93      Chicago   
8                           3.05                    2.46      Phoenix   
9                           1.20                    0.96     New York   
10                          1.93                    2.32      Chicago   

    User_Category  
0   Moderate User  
1      Heavy User  
2   Moderate User  
3   Moderate User  
4      Heavy User  
5      Light User  
6      Light User  
7      Heavy User  
8      Heavy User  
9   Moderate User  
10  Moderate User

In [62]:

import numpy as np

# Contoh DataFrame dengan 1000 baris dan 11 kolom
data = {
    "User_ID": range(1, 1001),
    "Daily_Screen_Time_Hours": np.random.randint(1, 12, 1000),
    "User_Category": np.random.choice(["Light User", "Moderate User", "Heavy User"], 1000),
    "Age": np.random.randint(18, 60, 1000),
    "Gender": np.random.choice(["Male", "Female"], 1000),
    "Location": np.random.choice(["City", "Suburb", "Rural"], 1000),
    "App_Usage_Time": np.random.randint(0, 10, 1000),
    "Browsing_Time": np.random.randint(0, 10, 1000),
    "Gaming_Time": np.random.randint(0, 10, 1000),
    "Streaming_Time": np.random.randint(0, 10, 1000),
    "Online_Shopping_Time": np.random.randint(0, 10, 1000),
}

df = pd.DataFrame(data)

# **1. Sampling** (mengambil subset dari data)
df_sample = df.sample(frac=0.5, random_state=42)  # Ambil 50% data secara acak

# **2. Dimensionality Reduction** (hapus fitur yang kurang penting)
df_reduced = df.drop(columns=["Browsing_Time", "Gaming_Time"])  # Hapus 2 kolom

# **3. Data Aggregation** (buat kolom total screen time)
df["Total_Screen_Time"] = (
    df["App_Usage_Time"] + df["Streaming_Time"] + df["Online_Shopping_Time"]
)

# **4. Data Compression** (konversi tipe data untuk hemat memori)
df["User_ID"] = df["User_ID"].astype("int16")
df["Age"] = df["Age"].astype("int8")

# Cek ukuran dataset setelah reduksi
print("Ukuran dataset awal:", df.memory_usage(deep=True).sum(), "bytes")
print("Ukuran dataset setelah reduksi:", df_reduced.memory_usage(deep=True).sum(), "bytes")



Ukuran dataset awal: 251123 bytes
Ukuran dataset setelah reduksi: 240123 bytes


**Penjelasan Output**

Ukuran dataset awal: 251123 bytes

Ini adalah total penggunaan memori dataset sebelum reduksi.

Ukuran dataset setelah reduksi: 240123 bytes

Setelah menghapus beberapa kolom (Browsing_Time, Gaming_Time), ukuran dataset berkurang.



In [69]:
print(df_reduced.head()) # Menampilkan dataset setelah pengurangan kolom


   User_ID  Daily_Screen_Time_Hours  User_Category  Age  Gender Location  \
0        1                       11     Heavy User   25  Female   Suburb   
1        2                        1     Light User   50    Male    Rural   
2        3                        9     Light User   18    Male     City   
3        4                        1  Moderate User   47    Male    Rural   
4        5                        6     Heavy User   19    Male     City   

   App_Usage_Time  Streaming_Time  Online_Shopping_Time  
0               5               6                     1  
1               7               0                     4  
2               5               7                     5  
3               2               9                     4  
4               4               6                     6  
